backend/src/api/query.rs

```diff
@@ -70,15 +70,18 @@ impl Query {
 
     /// Returns the current user.
     fn current_user(context: &Context) -> Option<&User> {
-        context.user.as_ref()
+        match context.authn {
+            AuthContext::User(ref user) => Some(user),
+            _ => None,
+        }
```

Comment:
As already mentioned, since [RFC 2005](https://rust-lang.github.io/rfcs/2005-match-ergonomics.html), `ref` is almost never needed anymore and it's more idiomatic to instead do this:

```suggestion
        match &context.authn {
            AuthContext::User(user) => Some(user),
            _ => None,
        }
```